### PR TITLE
Improve text fit in treemap trace

### PR DESCRIPTION
**Pull request by archmoj**
This PR:
- improves text fit scale when adjusted to `left` or `right` (by removing extra pad) 
- keeps constant text anchor positions when `unifromtext` rescales the sizes
- avoids hidden elements (namely when using `maxdepth`) to impact uniform text size

@plotly/plotly_js

### CODE REVIEW

File: src/traces/bar/plot.js [hunk 1]
```diff
@@ -479,9 +482,15 @@ function toMoveInsideBar(x0, x1, y0, y1, textBB, opts) {
     }
 
     // compute text and target positions
-    var textX = (textBB.left + textBB.right) / 2;
+    var textX = (
+        (textBB.left + 0 * TEXTPAD) * toLeft +
+        (textBB.right - 0 * TEXTPAD) * toRight
```
Comment (etpinard) [inline src/traces/bar/plot.js]: Do we need to make the `0 * TEXTPAD` explicit? 

I'd prefer

```
        textBB.left * toLeft +
        textBB.right * toRight
```

**Comment by etpinard** [comment]
Great work @archmoj The updated baselines look very nice! I made one comment.

File: src/traces/bar/plot.js (hunk 1 above)
Comment (archmoj) [reply src/traces/bar/plot.js]: Good eye. 
Thanks. I totally forgot to clean that one.
Fixed by https://github.com/plotly/plotly.js/pull/4469/commits/7098a97097e9e4f2541eb1d2f348ba801b3814a9.

Comment (etpinard) [reply src/traces/bar/plot.js]: Thanks!!

**Comment by etpinard** [comment]
:dancer: - thanks!

**Comment by etpinard** [comment]
(I didn't mean to close this PR, I clicked on the wrong button, sorry)

**Pull request merged**